java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/WorkspaceServiceImpl.java

```diff
@@ -401,11 +403,15 @@ public CompletableFuture<Object> executeCommand(ExecuteCommandParams params) {
                     LOG.log(Level.INFO, "Project {2}: {0} test roots opened in {1}ms", new Object[] { testRoots.size(), (System.currentTimeMillis() - t), file});
                     BiFunction<FileObject, Collection<TestMethodController.TestMethod>, Collection<TestSuiteInfo>> f = (fo, methods) -> {
                         String url = Utils.toUri(fo);
+                        Project owner = FileOwnerQuery.getOwner(fo);
+                        String moduleName = owner != null ? ProjectUtils.getInformation(owner).getDisplayName(): null;
```

Comment:
Careful here. Check with customized gradle and maven project (display) names (different from artfiact names) that it works well.

Reply:
Yes. For the maven we will have `parentName-artifactId`, and for the gradle it's `parentName:moduleName`